lib/queue.rb

```diff
@@ -1,28 +1,37 @@
 class Queue
+  attr_accessor :front, :rear, :store
+  attr_reader :size
 
-  def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+  def initialize(size=20)
+    @size = size
+    @store = Array.new()
```

Comment:
You should probably start the array at the given size

```suggestion
    @store = Array.new(size)
```